payment-in-stablecoins/README.md

```diff
@@ -0,0 +1,167 @@
+- `function ethNeeded(uint additionalUsdAmount, uint bufferFactor)` returns the amount of additional ETH needed in the contract to buy the USD backing needed for additionalUsdAmount + any unbacked minted NOU; `bufferFactor` is a volatility buffer scalar, e.g. when set to 2 the contract will ask for ETH with 2 times the USD value it needs to buy
+  - `bufferFactor` should have a hard-coded lower bound, e.g. 1.2 for 20% over-funding
+
+##### `sellStablecoin` transaction
+
+- `function sellStablecoin(uint amount)` sell this contract stablecoins in exchange for ETH at the exchange rate determined by the `price` function
+
+Rules
+
+- `amount` cannot exceed `stablecoinAmountNeeded()`
+- `msg.sender` must approve this contract to spend `amount` of stablecoin, which it will transfer to itself
+- should revert if `this.balance < amount / price()`
+
+##### `redeem` transactions
+
+Non-strict:
+
+- `function redeem(uint amount)` redeem `amount` stablecoins in exchange for the same amount of NOU tokens you own
+- `function redeem()` same as above, using `msg.sender`'s NOU balance as the amount
+
+Strict:
+
+- `function redeemExact(uint amount)` redeem `amount` stablecoins in exchange for the same amount of NOU tokens you own
+- `function redeemExact()` same as above, using `msg.sender`'s NOU balance as the amount
+
+Rules
+
+- `iouToken.balanceOf(msg.sender) > 0` must be true
+- `amount` cannot exceed `iouToken.balanceOf(msg.sender)`
+- must call `iouToken.burn` with the amount of stablecoins sent to `msg.sender`
+- `redeemExact` functions should revert if `amount > stablecoin.balanceOf(this)`
+
+##### `mintIOUs` transaction
+
+- `function mintIOUs(address to, uint amount) payable` mints `amount` new `iouToken`s to `to`, and accepts ETH payments, allowing `NounsDAOExecutor` to provide ETH funding for buying `amount` stablecoins later
+
+Rules:
+
+- `msg.sender == admin`
+- `msg.value >= amount * FundingRate`
+
+##### Admin transactions
+
+- `function withdrawETH()`
+- `function withdrawStablecoin()`
+- `function setBotIncentiveBPs(uint newBotIncentiveBPs)`
+- `function setAdmin(address newAdmin)`
+- `function setBaselineStablecoinAmount(uint newAmount)`
+- `function setPriceFeed(address newPriceFeed)`
+
+Rules:
+
+- `msg.sender == admin`
+
+##### `receive` transaction
+
+- This contract accepts ETH, allowing the DAO to top off its ETH balance without having to mint iou tokens
+
+Rules:
+
+- should not revert
+
+#### `IPriceFeed` interface
+
+The interface `NounsStablecoinPayments` uses to fetch stablecoin/ETH prices.
+
+- `function price()` returns the latest price
+
+#### `PriceFeed` contract
+
+`PriceFeed` is the first implementation of `IPriceFeed`. It uses Chainlink as its only oracle, to avoid the extra gas costs of using multiple oracles, under the assumption that Chainlink is highly available.
+
+##### Constants
+
+- `TIMEOUT` the maximum oracle refresh lag considered acceptable
```

Comment:
Borrowing from the YFI contract, `STALE_AFTER` is a bit more expressive.

Reply:
fixed